go/vt/sqlparser/ast.go

```diff
@@ -499,22 +499,25 @@ func (node *Set) WalkSubtree(visit Visit) error {
 }
 
 // DDL represents a CREATE, ALTER, DROP or RENAME statement.
-// Table is set for AlterStr, DropStr, RenameStr.
+// Table is set for AlterStr, DropStr, RenameStr, ReorganizeStr
 // NewName is set for AlterStr, CreateStr, RenameStr.
 type DDL struct {
-	Action    string
-	Table     TableName
-	NewName   TableName
-	IfExists  bool
-	TableSpec *TableSpec
+	Action               string
+	Table                TableName
+	NewName              TableName
+	IfExists             bool
+	TableSpec            *TableSpec
+	PartitionName        ColIdent
```

Comment:
Sounds like a good idea. I think it should be a trivial change: on line 919 of sql.y, build an AlterSpec that combines the two variables. Or, maybe I'm missing something.

I agree with deferring the interface. That can always be added when needed.